### PR TITLE
Fix autoschedule bugs

### PR DESCRIPTION
**Pull request by oscarsu97**
* Fix autoschedule message usage as well.
* Change time to compare by LocalTime instead of just integer.
* Sets the duration to be more than zero and less than 24h in Duration class

### CODE REVIEW

File: src/main/java/seedu/planner/model/field/NameAndTagWithTime.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.planner.logic.commands.util.CommandUtil;
+import seedu.planner.model.tag.Tag;
+
+/**
+ * Represents a Name or Tag with a timing tagged to it.
+ */
+public class NameAndTagWithTime {
```
Comment (1nefootstep) [inline src/main/java/seedu/planner/model/field/NameAndTagWithTime.java]: Call it NameOrTagWithTime instead

File: src/main/java/seedu/planner/model/field/NameAndTagWithTime.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    private final Name name;
+    private final Tag tag;
+    private final LocalTime time;
+
+    public NameAndTagWithTime(Name name, Tag tag, LocalTime time) {
```
Comment (1nefootstep) [inline src/main/java/seedu/planner/model/field/NameAndTagWithTime.java]: You can try having 2 extra constructors that allow you just have name, time arguments or tag, time arguments

File: src/main/java/seedu/planner/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -73,24 +71,34 @@ public static Name parseName(String name) throws ParseException {
...
 
     /**
-     * Parses a {@code String argValue} into a {@code NameWithTime}.
+     * Parses a {@code String argValue} into a {@code NameAndTagWithTime}.
      * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code phone} is invalid.
      */
-    public static NameWithTime parseNameWithTime(String argValue) throws ParseException {
+    public static NameOrTagWithTime parseNameAndTagWithTime(String argValue, Prefix prefix) throws ParseException {
```
Comment (1nefootstep) [inline src/main/java/seedu/planner/logic/parser/ParserUtil.java]: Rename

File: src/main/java/seedu/planner/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -73,24 +71,34 @@ public static Name parseName(String name) throws ParseException {
...
+
+        assert args.length > 0;
+
+        if (args.length > 1) {
+            parsedTime = parseTime(args[args.length - 1]);
+
+            String trimmedArg = trimmedArgValue.substring(0, trimmedArgValue.length() - 5);
+            if (prefix.equals(PREFIX_NAME)) {
+                return new NameOrTagWithTime(new Name(trimmedArg), parsedTime);
+            } else {
```
Comment (1nefootstep) [inline src/main/java/seedu/planner/logic/parser/ParserUtil.java]: Is there a possibility that another prefix gets into this line of code? I don't think so but you can choose to code defensively here.

Comment (oscarsu97) [reply src/main/java/seedu/planner/logic/parser/ParserUtil.java]: okay will take note of it thank you

File: src/main/java/seedu/planner/logic/parser/ParserUtil.java (hunk 3 above)
Comment (oscarsu97) [reply src/main/java/seedu/planner/logic/parser/ParserUtil.java]: okay thank for spotting

**Comment by oscarsu97** [comment]
Closes #195 #181 #160 #157 #153

**Pull request merged**